packages/playground/_stories/main/Badge/Badge.stories.ts

```diff
@@ -1,75 +1,79 @@
 import { html } from "lit-html";
-import type { Meta, StoryFn } from "@storybook/web-components";
-
+import { ifDefined } from "lit-html/directives/if-defined.js";
+import { unsafeHTML } from "lit-html/directives/unsafe-html.js";
+import type { Meta } from "@storybook/web-components";
 import argTypes, { componentInfo } from "./argTypes.js";
 import type { StoryArgsSlots } from "./argTypes.js";
 import type { UI5StoryArgs } from "../../../types.js";
-
 import { DocsPage } from "../../../.storybook/docs";
-
-// @ts-ignore
 import type Badge from "@ui5/webcomponents/dist/Badge.js";
 
+import { Basic as Icon } from "../Icon/Icon.stories.js";
+
 const component = "ui5-badge";
 
 export default {
-    title: "Main/Badge",
-    component,
-    parameters: {
-        docs: {
-          page: DocsPage({ ...componentInfo, component })
-        },
-    },
-    argTypes,
+	title: "Main/Badge",
+	component,
+	parameters: {
+		docs: {
+			page: DocsPage({ ...componentInfo, component })
+		},
+	},
+	argTypes,
 } as Meta<Badge>;
 
-const Template: UI5StoryArgs<Badge, StoryArgsSlots> = (args) => html`<div></div>`;
+const Template: UI5StoryArgs<Badge, StoryArgsSlots> = (args) => {
+	return html`
+<ui5-badge
+	color-scheme="${ifDefined(args.colorScheme)}"
+	style="${ifDefined(args.style)}"
+>
+	${args.icon ? Icon({ name: args.icon, slot: "icon" }) : null}
```

Comment:
As this call the Icon story is defined to expect 2 arguments I guess the TS support of this resuing technic is not well supported yet. 
We can define the template in the following way in order to render a default icon slot and reuse the args of the Basic icon story:

```js
${args.icon ? html`<ui5-icon name="${Icon?.args?.name}"></ui5-icon>` : null}
```

Reply:
I changed the approach as you suggested - we now supply the full ui5-icon markup, so that it show up as as HTML in the slots section for 'icon'.